Source/WebCore/platform/graphics/ComplexTextController.cpp

```diff
@@ -371,9 +343,12 @@ void ComplexTextController::collectComplexTextRuns()
     const Font* synthesizedFont = nullptr;
     const Font* smallSynthesizedFont = nullptr;
 
+    CachedTextBreakIterator graphemeClusterIterator(m_run.text(), TextBreakIterator::Mode::Character, m_font.fontDescription().computedLocale());
```

Comment:
While we are always using 16bit `UChar` iterator (by converting 8bit string to 16bit string), here we are passing StringView, which can be 8bit or 16bit. Don't we need to pass the same source to `curr` and `end`?

Reply:
`curr` and `end` are parallel to the `CachedTextBreakIterator`. They don't have to point to the same thing.

That being said, upon further inspection, I don't think we need `cp` or `curr` at all now. But that's a significant refactoring, so I'd like to do that in a follow-up patch.